### PR TITLE
Custom field "list of images" doesn't show directories properly

### PR DESCRIPTION
**Pull request by eshiol**
Pull Request for Issue #16708. replace PR #17125

### Summary of Changes
This patch fixes JFormFieldFolderList. See: #16708

### Testing Instructions
#16708 This patch should fix this behavior

### Expected result
Field "list of images" works correctly.

### CODE REVIEW

**Comment by Septdir** [comment]
@eshiol Maybe add trim?  
Perhaps someone will write the path `/images/path`

File: libraries/joomla/form/fields/folderlist.php [hunk 1]
```diff
@@ -186,7 +186,7 @@ protected function getOptions()
 
 		$path = $this->directory;
 
-		if (!is_dir($path))
+		if (!is_dir($path) || is_dir(JPATH_ROOT . '/' . $path))
```
Comment (laoneo) [inline libraries/joomla/form/fields/folderlist.php]: We should return when `if (!is_dir($path) && !is_dir(JPATH_ROOT . '/' . $path))` and not let it through as `JFolder::folders` will then return false anyway.

**Comment by brianteeman** [comment]
@eshiol please can you update with the suggestion from @laoneo

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 15bc34408860d0d8c5799981735f852a848dcbaf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20294">issues.joomla.org/tracker/joomla-cms/20294</a>.</sub>

**Comment by anibalsanchez** [comment]
The patch can't be applied.

The file marked for modification does not exist: libraries/joomla/form/fields/folderlist.php

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20294">issues.joomla.org/tracker/joomla-cms/20294</a>.</sub>

**Comment by brianteeman** [comment]
@anibalsanchez again this is a Pull Request for staging not joomla 4

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 46b13ac67e174ca12904e78f70eb2e97644698f1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20294">issues.joomla.org/tracker/joomla-cms/20294</a>.</sub>

**Comment by jwaisner** [comment]
@viocassel Please test latest version of PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20294">issues.joomla.org/tracker/joomla-cms/20294</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 46b13ac67e174ca12904e78f70eb2e97644698f1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20294">issues.joomla.org/tracker/joomla-cms/20294</a>.</sub>

**Comment by viocassel** [comment]
@jwaisner Done 🤘

File: libraries/joomla/form/fields/folderlist.php [hunk 2]
```diff
@@ -184,13 +184,20 @@ protected function getOptions()
 	{
 		$options = array();
 
-		$path = $this->directory;
+		$path = ltrim($this->directory, '/');
```
Comment (laoneo) [inline libraries/joomla/form/fields/folderlist.php:187]: Is this really needed?

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20294">issues.joomla.org/tracker/joomla-cms/20294</a>.</sub>

**Comment by luistar15** [comment]
[This change has broken some features.](https://github.com/joomla/joomla-cms/commit/10db7090f2113af971acced87d03726ca2345fe3#commitcomment-37901126)

**Pull request merged**